javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -36,11 +60,10 @@
     }
 
     static Object[] copy(Object arrayObject, int minLength) {
-        final Object[] array = (Object[]) arrayObject;
-        final int arrayLength = (array == null) ? 0 : array.length;
-        final int length = Math.max(arrayLength, minLength);
-        return (length == arrayLength) ? array.clone()
-                                       : arrayCopy(length, array, 0, 0, arrayLength);
+        final Object[] array = (arrayObject == null) ? EMPTY : (Object[]) arrayObject;
+        return (array.length >= minLength)
```

Comment:
empty can be called with `minLength` 10 also, resulting in a simple allocation, no copying.
This might be counter-intuitive, but allocation has to be in a single place, not spread all over the codebase.